### PR TITLE
Initial attempt to add repartitioning logic for large files.

### PR DESCRIPTION
**Pull request by sumabh-msft**

### CODE REVIEW

**Comment by sumabh-msft** [comment]
[like]  Suman Bhagavathula reacted to your message:
________________________________

File: DataProcessing/datax-host/src/main/scala/datax/processor/CommonProcessorFactory.scala [hunk 1]
```diff
@@ -706,7 +755,7 @@ object CommonProcessorFactory {
         val batchTimeStr = DateTimeUtil.formatSimple(batchTime)
         val batchLog = LogManager.getLogger(s"BatchProcessor-B$batchTimeStr")
 
-        groupPartitionProcessMetrics(pathsRDD, batchTime, batchInterval, outputPartitionTime, namespace, batchLog, metricLogger)
+        groupPartitionProcessMetrics(pathsRDD, batchTime, batchInterval, outputPartitionTime, namespace, batchLog, metricLogger, 0)
```
[the remark this answers is not in the document]
Comment (sumabh-msft) [reply DataProcessing/datax-host/src/main/scala/datax/processor/CommonProcessorFactory.scala]: updated.

File: DataProcessing/datax-core/pom.xml [hunk 2]
```diff
@@ -50,7 +50,7 @@ SOFTWARE
   </scm>
 
   <groupId>com.microsoft.datax</groupId>
-  <artifactId>datax-core_3.1_2.12</artifactId>
+  <artifactId>datax-core_3.1_2.13</artifactId>
   <version>1.2.17-SNAPSHOT</version>
```
Comment (rohit489) [inline DataProcessing/datax-core/pom.xml:54]: The version bump will be from 1.2.17-SNAPSHOT to 1.2.18-SNAPSHOT. The 2.12 is the scala version

Comment (sumabh-msft) [reply DataProcessing/datax-core/pom.xml]: fixed. thanks for the pointer.

**Pull request merged**